lib/web_ui/lib/src/engine/dom_renderer.dart

```diff
@@ -151,6 +157,10 @@ class DomRenderer {
   html.Element? get glassPaneElement => _glassPaneElement;
   html.Element? _glassPaneElement;
 
+  /// The ShadowRoot of the [glassPaneElement].
+  html.ShadowRoot? get glassPaneShadow => _glassPaneShadow;
+  html.ShadowRoot? _glassPaneShadow;
```

Comment:
I'm having trouble finding a base class that doesn't make me pepper the rest of the code with casts, and that works at the same time with ShadowRoot and Document.

(Maybe the casts *are* necessary for this to support both root types!)

Reply:
@ditman What about creating a wrapper class that takes the element in its constructor and then only does the cast once. Now, you can add members for all properties you need (like an interface) and they are type safe because you handle both cases in the wrapper.

Reply:
(you can do it in a way that prevents all casts)

```suggestion
factory _Wrapper(element) {
  return _Wrapper(
    element is ShadowRoot ? element : null,
    element is Document ? element : null,
  );
}

final ShadowRoot? shadowRoot;
final Document? document;

T get memberA => shadowRoot?.memberA ?? document!.memberA;
```

Forgive the syntax, typed it on my phone.

Reply:
@creativecreatorormaybenot yes! I thought of something similar maybe with extension methods. But yeah, this is probably better a TODO for the next version of this :P

Reply:
This ended up happening, here: https://github.com/flutter/engine/pull/26647